src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessarysemicolonaftertypedeclaration/InputUnnecessarySemicolonAfterTypeDeclaration.java

```diff
@@ -0,0 +1,45 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessarysemicolonaftertypedeclaration;
+
+/**
+ * Config = default
+ */
+public class InputUnnecessarySemicolonAfterTypeDeclaration {
+
+    enum innerEnum {
+
+    }; // OK, nested enum
+
+    @interface innerAnnotation {
+
+    }; // OK, nested annotation
+
+    interface innerInterface {
+
+    }; // OK, nested interface
+
+    class innerClass {
+
+    };  // OK, nested class
+
+}; // violation
+
+enum e {
+
+}; // violation
+
+@interface an {
+
+}; // violation
+
+interface i {
+
+}; // violation
+
+
+enum okEnum {}
```

Comment:
Just a note: these classes are needed for 100% coverage?

Reply:
We need 2 of them - one where next sibling is not semicolon, and one where next sibling is null.
I decided to keep all 4 possible type declarations.

Reply:
@pbludov , the more test cases the better, we should not reply on exact implementation of Check.